### PR TITLE
Removed Conditions trait from cluster. Improved label selectors.

### PR DESCRIPTION
**Pull request by maltesander**
## Description
- Removed conditions. 
- Sorted methods. 
- Bumped tokio to 1.10.
- Added list_config_maps. Renamed "get" to "list".

## Review Checklist
- [ ] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)

### CODE REVIEW

File: src/operator/setup.rs [hunk 1]
```diff
@@ -184,6 +113,103 @@ where
...
+
+    /// Returns all available nodes in the cluster. Can be used to determine the expected pods
+    /// for tests (depending on the custom resource)
+    pub fn list_nodes(&self) -> Vec<Node> {
+        self.client
+            .list_labeled::<Node>("kubernetes.io/arch=stackable-linux")
+            .items
+    }
+
+    /// List all available pods in the cluster via the name label.
```
Comment (soenkeliebau) [inline src/operator/setup.rs:149]: 'Available' is misleading in this context I think. This refers to the existing pods that have been created in kube-apiserver, correct?

File: src/operator/setup.rs [hunk 2]
```diff
@@ -184,6 +113,103 @@ where
...
+        let cluster_name = if self.cluster.is_some() {
+            T::name(self.cluster.as_ref().unwrap())
+        } else {
+            "<not-found>".to_string()
+        };
+        format!("[{}/{}] {}", T::kind(&()), cluster_name, message)
+    }
+
+    /// This creates a kube client and should be executed at the start of every test.
+    pub fn new(options: TestClusterOptions, timeouts: TestClusterTimeouts) -> Self {
```
Comment (soenkeliebau) [inline src/operator/setup.rs:180]: nit: I would prefer this to be at the top of the impl block.

**Review by soenkeliebau — CHANGES_REQUESTED** [review]
Left some comments

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -4,11 +4,20 @@
...
+
+=== Added
+* list_config_maps via label selector ({21}).
+* list_pods now has another label for cluster name ({21}).
 
 === Changed
 * Dependency `kube` set to version `0.58` ({15}).
 * Added operator checks for pod version (label) and pod creation timestamps ({19}).
 * Improved operator logging output (messages are prefixed with [<Kind>/<Name>]) ({19}).
+* Dependency `tokio` set to version `1.10` ({21}).
```
[the remark this answers is not in the document]
Comment (siegfriedweber) [reply CHANGELOG.adoc:17]: Today I discovered how Kubernetes lists its dependency changes: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.22.md#dependencies-1

It is just `<dependency>: <old version> → <new version>`.

[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc:17]: I like that. Now we only provide the new version. Should we try that or are there any other proposals? @siegfriedweber @soenkeliebau @lfrancke @razvan

File: CHANGELOG.adoc [hunk 4]
```diff
@@ -4,11 +4,20 @@
 
 :15: https://github.com/stackabletech/integration-test-commons/pull/15[#15]
 :19: https://github.com/stackabletech/integration-test-commons/pull/19[#19]
+:21: https://github.com/stackabletech/integration-test-commons/pull/21[#21]
+
+=== Added
+* list_config_maps via label selector ({21}).
+* list_pods now has another label for cluster name ({21}).
```
[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc:11]: done

File: CHANGELOG.adoc [hunk 5]
```diff
@@ -4,11 +4,20 @@
...
 
 === Changed
 * Dependency `kube` set to version `0.58` ({15}).
 * Added operator checks for pod version (label) and pod creation timestamps ({19}).
 * Improved operator logging output (messages are prefixed with [<Kind>/<Name>]) ({19}).
+* Dependency `tokio` set to version `1.10` ({21}).
+* All "get_" methods renamed to "list_" ({21}).
```
[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc:18]: done

File: CHANGELOG.adoc [hunk 6]
```diff
@@ -4,11 +4,20 @@
...
+* Sorted cluster methods alphabetically ({21}).
+* Renamed `pod_name_label` to `cluster_type` ({21}).
```
[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc:20]: done

File: src/operator/setup.rs [hunk 7]
```diff
@@ -184,6 +113,103 @@ where
...
+        ));
+
+        labels.extend(additional_labels);
+
+        self.client
+            .list_labeled::<ConfigMap>(&labels.join(","))
+            .items
+    }
+
+    /// Returns all available nodes in the cluster. Can be used to determine the expected pods
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/operator/setup.rs:141]: I improved it

File: src/operator/setup.rs (hunk 1 above)
Comment (maltesander) [reply src/operator/setup.rs:149]: done

File: src/operator/setup.rs (hunk 2 above)
Comment (maltesander) [reply src/operator/setup.rs:180]: done

File: CHANGELOG.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply CHANGELOG.adoc:17]: I'd be happy to do it like that.

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply CHANGELOG.adoc:17]: Decided as group in the daily on 31.08.2021 to use Kubernetes notation.

**Comment by soenkeliebau** [comment]
Just that one comment about notation for dependency updates left to do, after that happy to approve.

File: CHANGELOG.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (maltesander) [reply CHANGELOG.adoc]: done

**Review by soenkeliebau — APPROVED** [review]
LGTM

**Pull request merged**